src/rest-api/content-helper/class-endpoint-smart-linking.php

```diff
@@ -1,103 +1,124 @@
  *
- * Provides endpoints for managing smart links.
+ * Provides an endpoint for generating smart links for the given content.
  *
- * @since 3.16.0
+ * @since 3.17.0
  */
-class Smart_Linking_Endpoint extends Base_Endpoint {
+class Endpoint_Smart_Linking extends Base_Endpoint {
+	use Content_Helper_Feature;
 
 	/**
-	 * The endpoint base path.
+	 * The Suggest Linked Reference API instance.
 	 *
-	 * @since 3.16.0
-	 * @var string
+	 * @since 3.17.0
+	 *
+	 * @var Suggest_Linked_Reference_API $suggest_linked_reference_api
 	 */
-	protected const ENDPOINT = '/smart-linking';
+	private $suggest_linked_reference_api;
 
 	/**
-	 * Returns whether the endpoint is available for access by the current
-	 * user.
+	 * Initializes the class.
 	 *
-	 * @since 3.16.0
+	 * @since 3.17.0
 	 *
-	 * @param WP_REST_Request|null $request The request object.
-	 * @return bool
+	 * @param Content_Helper_Controller $controller The content helper controller.
 	 */
-	public function is_available_to_current_user( $request = null ): bool {
-		$post_id = false;
-		if ( $request instanceof WP_REST_Request ) {
-			$temp_post_id = $request->get_param( 'post_id' );
-			if ( is_numeric( $temp_post_id ) ) {
-				$post_id = intval( $temp_post_id );
-			}
-		}
-
-		$can_access_pch = Permissions::current_user_can_use_pch_feature(
-			'smart_linking',
-			$this->parsely->get_options()['content_helper'],
-			$post_id
-		);
+	public function __construct( Content_Helper_Controller $controller ) {
+		parent::__construct( $controller );
+		$this->suggest_linked_reference_api = new Suggest_Linked_Reference_API( $this->parsely );
+	}
 
-		// Check if the current user has the smart linking capability.
-		$has_capability = current_user_can(
-			// phpcs:ignore WordPress.WP.Capabilities.Undetermined
-			$this->apply_capability_filters(
-				Base_Endpoint::DEFAULT_ACCESS_CAPABILITY
-			)
-		);
+	/**
+	 * Returns the name of the endpoint.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @return string The endpoint name.
+	 */
+	public function get_endpoint_name(): string {
+		return 'smart-linking';
+	}
 
-		return $can_access_pch && $has_capability;
+	/**
+	 * Returns the name of the feature associated with the current endpoint.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @return string The feature name.
+	 */
+	public function get_pch_feature_name(): string {
+		return 'smart_linking';
 	}
 
 	/**
-	 * Registers the endpoints.
+	 * Registers the routes for the endpoint.
 	 *
-	 * @since 3.16.0
+	 * @since 3.17.0
 	 */
-	public function run(): void {
+	public function register_routes(): void {
 		/**
-		 * POST /smart-linking/url-to-post-type
-		 * Converts a URL to a post type.
+		 * GET /smart-linking/generate
+		 * Generates smart links for a post.
 		 */
-		$this->register_endpoint(
-			static::ENDPOINT . '/url-to-post-type',
-			'url_to_post_type',
-			array( 'POST' )
+		$this->register_rest_route(
+			'generate',
+			array( 'POST' ),
+			array( $this, 'generate_smart_links' ),
+			array(
```

Comment:
Seems we're missing a couple of `'require'` here?

Reply:
Not entirely sure I understand. The `text` parameter is the only mandatory parameter, the other ones - `max_links` and `url_exclusion_list` are not mandatory and have default values if omitted.

Reply:
Had that question because in other endpoints we do specify `'required' => false`. Let's keep to one convention by either removing those, or specifying `'required' => false` where needed.

Reply:
@vaurdan, maybe you've missed this one?